lib/services/classes.js

```diff
@@ -345,4 +345,70 @@ module.exports = class classesService extends Schmervice.Service {
       return [errorHandler(err), null];
     }
   }
+
+  async getIfStudentEnrolled(id, pathwayId) {
+    try {
+      let FinalResult = 'not_enrolled';
+      const [
+        errWhileFetchingByUserId,
+        classesRegisteredByUser,
+      ] = await this.getRecurringClassesIdByUserId(id);
+      if (classesRegisteredByUser != null) {
+        let i = 0;
+        while (i < classesRegisteredByUser.length) {
+          let result = true;
+          // eslint-disable-next-line
+          const [errWhileFetching, classDetails] = await this.getClassById(
+            classesRegisteredByUser[i]
+          );
+          if (classDetails.recurring_id) {
+            // eslint-disable-next-line
+            const [errInFetching, allRecurringClasses] = await this.getClassesIdByRecurringId(
+              classDetails.recurring_id
+            );
+            // eslint-disable-next-line
+            allRecurringClasses.forEach((ClassRegisterdId) => {
+              if (classesRegisteredByUser.indexOf(ClassRegisterdId) > -1) {
+                classesRegisteredByUser.splice(
+                  classesRegisteredByUser.indexOf(ClassRegisterdId),
+                  1
+                );
+              } else {
+                result = false;
+                i += 1;
+                // eslint-disable-next-line
+                return;
+              }
+            });
+            if (result) {
```

Comment:
I'm not sure if you saw my comment (https://github.com/navgurukul/sansaar/pull/489/files#r792184645) because I wrote it after the PR was merged, but I think the code that was replicated here can produce erroneous outputs in certain instances.  Also, because of the code duplication, I'd suggest factoring out the common code in a function that could be used by both.  Otherwise, as is the case here, a potential error in logic must be corrected in multiple places.  
```suggestion
            result = allRecurringClasses.every((ClassRegisteredId) => {
              const classIndex = classesRegisteredByUser.indexOf(ClassRegisteredId);
              return classIndex > -1 && !!classesRegisteredByUser.splice(classIndex, 1);
            });
            if (!result) {
              i++;
            } else {
```